lib/components/ConnectModal/index.tsx

```diff
@@ -13,26 +30,35 @@ interface Props {
   };
   buttonText: string;
   buttonClassName?: string;
-  buttonVariant?: "lightGray" | "gray" | "blue" | "red" | "transparent" | null;
   onClose?: () => void;
+  showModal: boolean;
 }
 
 export const ConnectModal = (props: Props) => {
-  const [showModal, setShowModal] = useState(false);
   const [step, setStep] = useState<"connect" | "error" | "loading">("connect");
-  const { address, connect, disconnect, isConnected } = useWeb3();
+  const { connect, toggleModal } = useWeb3();
+  const focusTrapRef = useFocusTrap();
+  const [showMetamask, setShowMetamask] = useState(false);
+  const [showBrave, setShowBrave] = useState(false);
 
-  const buttonVariant = props.buttonVariant ?? null;
+  useEffect(() => {
+    if (window.ethereum && (window.ethereum as any).isBraveWallet) {
```

Comment:
I just noticed these don't need to live in state because `window.ethereum` doesn't change at runtime.
they can just be variables.

* [ ] Assume window can also be undefined (server render).
* [ ] Also help typescript out a bit:

```ts
import { providers } from "ethers";
type WindowEthereum = providers.ExternalProvider && { isBraveBrowser: boolean }

const eth = window?.ethereum as WindowEthereum | undefined;
const showBrave = eth?.isBraveWallet;
const showMetamask = eth?.isMetamask;
const showCoinbaseWallet = eth?. isCoinbaseWallet;
```

Also coincidentally yapper reported a coinbase wallet issue a few hours ago and the bug is right here in this codeblock. See ticket (high priority) https://github.com/KlimaDAO/klimadao/issues/881

Reply:
updated. eth needs to be in state because we need to access window in a useEffect but the wallet checks are out of state now.